src/Sql/TableIdentifier.php

```diff
@@ -45,14 +43,17 @@ public function __construct($table, $schema = null)
         if (null === $schema) {
             $this->schema = null;
         } else {
-            if (! (is_string($schema) || is_callable([$schema, '__toString']))) {
+            if (! (is_string($schema) || is_array($schema) || is_callable([$schema, '__toString']))) {
                 throw new Exception\InvalidArgumentException(sprintf(
-                    '$schema must be a valid schema name, parameter of type %s given',
+                    '$schema must be a valid schema name or path in form of array, parameter of type %s given',
                     is_object($schema) ? get_class($schema) : gettype($schema)
                 ));
             }
 
-            $this->schema = (string) $schema;
+            if (! is_array($schema)) {
```

Comment:
probably check for callable `[$schema, '__toString']` only to avoid re-cast $schema that already string ?

```php 
if (is_callable([$schema, '__toString'])) {
```